napari/_app_model/_app.py

```diff
@@ -27,20 +66,125 @@ def __init__(self, app_name=APP_NAME) -> None:
         # adopt asynchronous command execution.
         super().__init__(app_name, raise_synchronous_exceptions=True)
 
+        self._repeatable_actions: Set[str] = set()
+
         self.injection_store.namespace = _napari_names  # type: ignore [assignment]
         self.injection_store.register(
             providers=PROVIDERS, processors=PROCESSORS
         )
 
-        for action in chain(HELP_ACTIONS, LAYER_ACTIONS, VIEW_ACTIONS):
+        for action in chain(
+            HELP_ACTIONS,
+            IMAGE_ACTIONS,
+            LABELS_ACTIONS,
+            LAYER_ACTIONS,
+            POINTS_ACTIONS,
+            SHAPES_ACTIONS,
+            VIEW_ACTIONS,
+            VIEWER_ACTIONS,
+        ):
             self.register_action(action)
 
+        # re-register with action_manager shim for keybindings
+        for keymapprovider, actions in (
+            (ViewerModel, VIEWER_ACTIONS),
+            (Image, IMAGE_ACTIONS),
+            (Labels, LABELS_ACTIONS),
+            (Points, POINTS_ACTIONS),
+            (Shapes, SHAPES_ACTIONS),
+        ):
+            for action in actions:
+                self._register_action_manager_shim(action, keymapprovider)
+
         self.menus.append_menu_items(SUBMENUS)
 
+        self.keybindings.registered.connect(populate_plugin_keymap)
+
+    def _register_action_manager_shim(self, action: Action, keymapprovider):
+        """Shim from app-model Action to action_manager for keybinding"""
```

Comment:
I think we should expand this docstring a bit to explain why we're performing this shimming and when (if ever) we can get rid of it.

Reply:
Yes we should be able to get rid of it eventually I think but I can make that clear in the doc string. The TODO just below kind of explains it but having it in the doc string makes sense and less likely to be forgotten.